lumispy/signals/common_luminescence.py

```diff
@@ -118,18 +119,26 @@ def scale_by_exposure(self, exposure=float("nan"), inplace=False):
 
         # Make sure exposure is given or contained in metadata
         if isnan(exposure):
-            # use nested_get from hyperspy when it is available
-            if self.metadata.has_item("Acquisition_instrument.CL.exposure"):
+            if self.metadata.has_item(
+                "Acquisition_instrument.Detector.integration_time"
+            ):
                 exposure = float(
-                    self.metadata.get_item("Acquisition_instrument.CL.exposure")
+                    self.metadata.get_item(
+                        "Acquisition_instrument.Detector.integration_time"
+                    )
                 )
-            elif self.metadata.has_item("Acquisition_instrument.CL.dwell_time"):
+            # following will work only from hyperspy v1.7
+            elif self.metadata.has_item("integration_time", full_path=False):
                 exposure = float(
-                    self.metadata.get_item("Acquisition_instrument.CL.dwell_time")
+                    self.metadata.get_item("integration_time", full_path=False)
                 )
+            elif self.metadata.has_item("exposure", full_path=False):
+                exposure = float(self.metadata.get_item("exposure", full_path=False))
+            elif self.metadata.has_item("dwell_time", full_path=False):
+                exposure = float(self.metadata.get_item("dwell_time", full_path=False))
```

Comment:
Would it more simple and readable to do something along the line of:
```python
if exposure is None:
   # exposure is not provided, try to get it from metadata
    exposure = self.metadata.get_item("Acquisition_instrument.Detector.integration_time")
if exposure is None:
    # integration_time is not in the metadata, raise an error
    raise ValueError("Provide `exposure` argument or set `integration_time` in the metadata.")
```

Why should other cases (integration time saved elsewhere in the metadata) be supported? The point of defining a metadata specification is avoid this kind of thing!

Reply:
> Why should other cases (integration time saved elsewhere in the metadata) be supported? The point of defining a metadata specification is avoid this kind of thing!

Because, even though we have a metadata convention, not all readers parse the metadata yet. But in fact, we should then search the `original_metadata` and not the `metadata`.

Reply:
The readers need to be fixed, then! :)
The motivation for the metadata specification is for user *and* hyperspy functionalities to know where to find relevant information and avoid using heuristic to find it which in my opinion is a bad practice and it needs to be avoided. I don't think, that this is explained in the hyperspy user guide, which would be good to do.

Considering that lumispy doesn't have a stable API at the moment, I would not worry too much about rough edges like this, where the expose needs to be passed explicitly and favour the use of good practice.